### ISSUE TITLE
Fix Vagrant ("bug@ITC")

### ISSUE THREAD

**Issue by rzuritamilla** [phenology/infrastructure#6]
IPv4 vs IPv6

**Comment by romulogoncalves** [comment]
The issue might be solved with the following check-in:
https://github.com/nlesc-sherlock/emma/commit/b94ec815936f1697f093a4e2b5782fc4c21d5593

It still needs to be tested at ITC Twente and if it solves the problem ported to this repository.

@rzuritamilla when we report bugs an issue should be enough. I think we should not define it as a task in the TODO list of a project. Otherwise, we end up with long list of tasks.

**Comment by romulogoncalves** [comment]
It seems for some public networks it is not possible to create set of machines with public IP and have them interconnected. The workaround for such issue is to set a private network instead of public network. To do that in the [Vagrantfile] the following lines should be replaced:
```
-  config.vm.network "public_network"
+ config.vm.network "private_network", type: "dhcp"
```

**Note:** with this option web-sites will not be available through normal IP resolution because the vagrant machines will be on a different network. Hence, it is necessary to forward ports. It is possible to ask vagrant to do such port forwarding. The following should be added to [Vagrantfile]
```
#auto_correct will make sure if the port is in use at the host, it picks another one. 
config.vm.network "forwarded_port", guest: 9091, host: 9091, host_ip: "127.0.0.1", auto_correct: true, id: "minio"
config.vm.network "forwarded_port", guest: 8080, host: 8080, host_ip: "127.0.0.1", auto_correct: true, id: "spark_master"
config.vm.network "forwarded_port", guest: 8000, host: 8000, host_ip: "127.0.0.1", auto_correct: true, id: "jupyterhub"
```

In case a cluster is already set vagrant reload will not be enough. We recommend a vagrant destroy and vagrant up. Now the JupyterHub portal will be available at *http://127.0.0.1:8000*.

**Comment by romulogoncalves** [comment]
It seems the issue is related with ITC Eduroam, it has a setting which does not allow a machine to request more than one IP.
